src/clients/Discourse.ts

```diff
@@ -295,10 +293,11 @@ export class Discourse extends API {
 
   constructor(baseUrl: string) {
     super(baseUrl)
-    this.auth = this.getCredentials()
+    this.auth = Discourse.getCredentials()
   }
 
-  private getCredentials() {
+  private static getCredentials() {
+    const DISCOURSE_API_KEY = requiredEnv('DISCOURSE_API_KEY')
```

Comment:
Isn't this already throwing an error if env is not present? If this is the case then condition in line 301 is not necessary anymore